keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactListener.kt

```diff
@@ -114,11 +108,12 @@ class ArtifactListener(
         repository.getAllArtifacts().forEach { artifact ->
           launch {
             val lastRecordedVersion = getLatestStoredVersion(artifact)
-            val latestVersion = when (artifact) {
-              is DebianArtifact -> getLatestDeb(artifact)?.let { "${artifact.name}-$it" }
-              is DockerArtifact -> getLatestDockerTag(artifact)
-              else -> error("Unrecognized artifact type: ${artifact.type}")
-            }
+            val artifactPublisher = artifactPublishers.supporting(artifact.type)
+            val deliveryConfig = artifact.deliveryConfigName
```

Comment:
This bit is repeated, it's probably worth pulling it into a function since it also involves error handling